### ISSUE TITLE
PLW0642 flags inplace operators called on self

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#12954]
```
ruff check --select="PLW0642"
```

```python
    def symmetric_difference_update(self, other: Iterable[T]) -> None:
         self ^= other  # type: ignore[operator, arg-type]
```
is errantly flagged by PLW0642, but I am in these case I'm calling an overloaded function `operator.ixor` on self, not trying to assign a new variable to it. The former would have an actual side affect here, while the later would not affect it at all (which this bug is trying to catch).

In this case, `other` is a set, and I can update typing if that the typing is what is confusing ruff, but I wanted to
version: `ruff 0.6.0`

**Comment by AlexWaygood** [comment]
It makes sense to me to exclude augmented assignments from this rule, and it would also mean we'd have more similar behaviour to pylint here. (Pylint seems to ignore augmented assignments for their version of this rule -- though they also ignore annotated assignments, which makes less sense to me.)

@diceroll123 -- was there a specific reason you decided to differ from pylint's behaviour here when you added this rule in https://github.com/astral-sh/ruff/pull/9267? :-)

**Comment by diceroll123** [comment]
Oversight on my part.

Is the desire to remove augmented assignments and keep annotated? 😄

**Comment by AlexWaygood** [comment]
> Is the desire to remove augmented assignments and keep annotated? 😄

Yeah, I think so!
